packages/drawer/rc/Drawer.js

```diff
@@ -0,0 +1,156 @@
+import { css } from 'glamor'
+import PropTypes from 'prop-types'
+import React, { forwardRef, useEffect, useImperativeHandle } from 'react'
+
+import filterReactProps from '@pluralsight/ps-design-system-filter-react-props'
+import { CaretDownIcon } from '@pluralsight/ps-design-system-icon'
+import { useTheme } from '@pluralsight/ps-design-system-theme'
+import { useCollapsible } from '@pluralsight/ps-design-system-collapsible/rc'
+
+import stylesheet from './css/index.js'
+
+import { DrawerContext, useDrawerContext } from './context.js'
+import { actionTypes, useDrawer } from './use-drawer.js'
+import { usePrevious } from './use-previous.js'
+import { combineFns, isDefined, isFunction } from './utils.js'
+
+const styles = {
+  head: (themeName, { isOpen }) =>
```

Comment:
for style args i like to do theme first, props seconds, and component state third